odl/solvers/nonsmooth/difference_convex.py

```diff
@@ -0,0 +1,243 @@
+# Copyright 2014-2018 The ODL contributors
+#
+# This file is part of ODL.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public License,
+# v. 2.0. If a copy of the MPL was not distributed with this file, You can
+# obtain one at https://mozilla.org/MPL/2.0/.
+
+"""Solvers for the optimization of the difference of convex functions.
+
+Collection of DCA and related methods which make use of structured optimization
+if the objective function can be written as a difference of two convex
+functions.
+"""
+
+from __future__ import print_function, division, absolute_import
+
+__all__ = ('dca', 'prox_dca', 'doubleprox_dc')
+
+
+def dca(x, g, h, niter, callback=None):
+    r"""Subgradient DCA of Tao and An.
+
+    This algorithm solves a problem of the form::
+
+        min_x g(x) - h(x),
+
+    where ``g`` and ``h`` are proper, convex and lower semicontinuous
+    functions.
+
+    Parameters
+    ----------
+    x : `LinearSpaceElement`
+        Initial point, updated in-place.
+    g : `Functional`
+        Convex part. Needs to provide a `Functional.convex_conj` with a
+        `Functional.proximal` factory.
```

Comment:
`gradient`

Reply:
I changed this. Maybe this is an artefact of diffing.